### PR TITLE
MAT-10061: update spring boot (from 3.5.14 to 4.0.7)

### PR DESCRIPTION
**Pull request by etan-sb**
## MADiE PR

Jira Ticket: [MAT-10061](https://jira.cms.gov/browse/MAT-10061)
(Optional) Related Tickets:

### Summary

# Spring Boot 4 + Jackson 3 Upgrade Notes — cql-library-service

Part of the fleet-wide SB 3.5.14 → 4.0.6 + Jackson 2 → 3 migration. Common patterns live in
`madie-fhir-service/SPRING_BOOT_4_UPGRADE.md`; this file records only what was specific to this repo.

## pom.xml
- Parent `3.5.14` → `4.0.6`. okta `3.0.7` → `3.1.0`.
- `madie-java-models` `0.9.6-SNAPSHOT` → `0.10.0-SNAPSHOT`; `madie-rest-commons` `1.0.6-SNAPSHOT`
  → `1.1.0-SNAPSHOT`.
- `spring-boot-starter-web` → `spring-boot-starter-webmvc`; added `spring-boot-restclient`,
  `spring-boot-starter-security-oauth2-resource-server`, `spring-boot-starter-security-oauth2-client`.
- Test deps: added `spring-boot-webmvc-test`; `spring-security-test` → `spring-boot-starter-security-test`.

## Source — Spring 7 / Security 7
- **`SecurityConfig` — first repo to need the full SS6 `.and()` → SS7 lambda-DSL rewrite.** Removed the
  six chained `.and()` calls; each section (`cors`/`csrf`/`authorizeHttpRequests`/`sessionManagement`/
  `oauth2ResourceServer`/`headers`) is now a `Customizer` lambda. Added `@EnableWebSecurity`. The
  `headers().xssProtection().and().contentSecurityPolicy("…")` became
  `headers(h -> h.xssProtection(withDefaults()).contentSecurityPolicy(csp -> csp.policyDirectives("…")))`.
- `UserServiceClientConfig`: removed the Jackson-2 `MappingJackson2HttpMessageConverter` surgery (only
  existed to bind the J2 ObjectMapper) → bare `new RestTemplateBuilder().additionalInterceptors(...).build()`.
  `RestTemplateBuilder` import `org.springframework.boot.web.client` → `org.springframework.boot.restclient`.
- `ErrorHandlingControllerAdvice`: `ErrorAttributes` import `org.springframework.boot.web.servlet.error`
  → `org.springframework.boot.webmvc.error`. (`ErrorAttributeOptions` **stays** on
  `org.springframework.boot.web.error`.)

## Source — Jackson 3
- `ElmTranslatorClient`: `com.fasterxml.jackson.databind.{JsonNode,ObjectMapper}` → `tools.jackson.databind`;
  `new ObjectMapper()` → `JsonMapper.builder().build()`. **`JsonNode.asText()` is strict in J3** (throws on
  missing/non-scalar) → `errorException.path("errorSeverity").asText()` → `.asString("")` (default-valued
  overload returns `""` instead of throwing, so the `.equals("Error")` comparison still works).
- `LibraryListDTO`: the databind-level `@JsonSerialize`/`@JsonDeserialize(using=VersionJsonSerializer…)`
  annotations move `com.fasterxml.jackson.databind.annotation` → `tools.jackson.databind.annotation`.
  (`@JsonInclude` is a core annotation → **stays** on `com.fasterxml.jackson.annotation`.)

## MongoDB property split (5 ymls)
- `spring.data.mongodb.{uri,database}` → `spring.mongodb.{uri,database}` in `application.yml`,
  `application-aws.yml`, `application-it.yml`, `application-local.yml`, and test `application-test.yml`.
  (`application-aws.yml` also had a misplaced `spring.data.database` → folded under `spring.mongodb`.)

## Tests
- `@WebMvcTest` package `org.springframework.boot.test.autoconfigure.web.servlet` →
  `org.springframework.boot.webmvc.test.autoconfigure` (both MvcTest files).
- **⚠️ `@Captor` no longer initialized under SB4.** With the old `@MockBean` removed in favor of
  `@MockitoBean` (bean-override), the bean-override mechanism does **not** initialize plain Mockito
  `@Captor`/`@Mock` fields the way the old `MockitoTestExecutionListener` did → `ArgumentCaptor` fields
  were `null` at runtime (NPE on `.capture()`). Fix: add `@ExtendWith(MockitoExtension.class)` to the test
  class (alongside the `@WebMvcTest`-supplied `SpringExtension`). MockitoExtension only manages the
  `@Captor`/`@Mock` fields it creates — it does not validate the `@MockitoBean` stubs, so no
  strict-stubbing fallout. (First repo in the fleet to hit this; expect it wherever a slice test uses
  `@Captor`/`@Mock`.)
- **`SerializationFeature.WRITE_DATES_AS_TIMESTAMPS` relocated in Jackson 3** → `DateTimeFeature`
  (`tools.jackson.databind.cfg.DateTimeFeature`); the mapper helper became
  `JsonMapper.builder().disable(DateTimeFeature.WRITE_DATES_AS_TIMESTAMPS).build()`. `JavaTimeModule`
  removed (jsr310 auto-registered). `JsonProcessingException` (checked) → `JacksonException` (unchecked);
  since it's unchecked, `throws` clauses that only covered it were dropped.
- **Jackson 3 alphabetical ordering** broke 4 exact-JSON-string assertions
  (`VersionJsonSerializerTest` CqlLibrary round-trip + 3 in `CqlLibraryControllerMvcTest`:
  `testGetLibraryUsage`, `testUpdateSharedLibraries`, `testUnshareLibraries`). Fixed the **tests**, not
  production DTOs, with `JSONAssert.assertEquals(expected, actual, JSONCompareMode.STRICT)` (qualified, not
  static-imported, to avoid clashing with the JUnit `assertEquals` static import already in the file).
  `JSONAssert.assertEquals` throws checked `org.json.JSONException` → declared on the JSON-comparing methods.

## Result
`mvn clean install` green — 491 tests, 0 failures.

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included and Code coverage is verified.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/test/java/gov/cms/madie/cqllibraryservice/utils/VersionJsonSerializerTest.java [hunk 1]
```diff
@@ -78,7 +80,7 @@ public void testDeSerializerHandlesNull() {
     Version output = null;
     try {
       objectMapper.readValue(json, Version.class);
-    } catch (JsonProcessingException ex) {
+    } catch (JacksonException ex) {
```
Comment (jkotanchik-SB) [inline src/test/java/gov/cms/madie/cqllibraryservice/utils/VersionJsonSerializerTest.java]: These catch statements can be removed since an exception should fail the test.

File: src/test/java/gov/cms/madie/cqllibraryservice/utils/VersionJsonSerializerTest.java [hunk 2]
```diff
@@ -89,7 +91,7 @@ public void testDeSerializerHandlesException() {
...
```
Comment (jkotanchik-SB) [inline src/test/java/gov/cms/madie/cqllibraryservice/utils/VersionJsonSerializerTest.java]: These catch statements can be removed since an exception should fail the test.

**Review by jkotanchik-SB — CHANGES_REQUESTED** [review]
Remove no-op `JacksonException` catch blocks in unit tests.

**Review by jkotanchik-SB — APPROVED** [review]
